contracts/contracts/layer_0/clearing/ClearingStorageWrapper2.sol

```diff
@@ -455,36 +464,43 @@ abstract contract ClearingStorageWrapper2 is
             _amount,
             _clearingOperation.expirationTimestamp,
             _clearingOperation.data,
-            _operatorData
+            _operatorData,
+            _thirdPartyType
         );
 
-        success_ = true;
+        _clearingStorage().clearingThirdParty[_from][
+            _clearingOperation.partition
+        ][IClearing.ClearingOperationType.Redeem][
+                clearingId_
+            ] = _thirdPartyAddress;
```

Comment:
Why we share things. crateClearing is one thing and thirdParty/allowance is another. This shouldn't be duplicated.